src/DynamoCore/Models/PresetsModel.cs

```diff
@@ -0,0 +1,190 @@
+                        var des = stateNode.GetAttribute("Description");
+                        var stateguidString = stateNode.GetAttribute("guid");
+
+                        Guid stateID;
+                        if (!Guid.TryParse(stateguidString, out stateID))
+                        {
+                            logger.LogError("unable to parse the GUID for preset state: " +name+ ", will atttempt to load this state anyway");
+                        }
+
+                        var nodes = new List<NodeModel>();
+                        var deserialzedNodes = new List<XmlElement>();
+                        //now find the nodes we're looking for by their guids in the loaded nodegraph
+                        //it's possible they may no longer be present, and we must not fail to set the
+
+                        //iterate each actual saved nodemodel in each state
+                        foreach (XmlElement node in stateNode.ChildNodes)
+                        {
+                            var nodename = stateNode.GetAttribute("nickname");
+                            var guidString = node.GetAttribute("guid");
+                            Guid nodeID;
+                            if (!Guid.TryParse(guidString, out nodeID))
+                            {
+                                logger.LogError("unable to parse GUID for node " +nodename );
+                                continue;
+                            }
+
+                            var nodebyGuid = nodegraph.Nodes.Where(x => x.GUID == nodeID).ToList();
+                            if (nodebyGuid.Count > 0)
+                            {
+                                nodes.Add(nodebyGuid.First());
+                                deserialzedNodes.Add(node);
+                            }
+                            else
+                            {   //add the deserialized version anyway so we dont lose this node from all states.
+                                deserialzedNodes.Add(node);
+                                logger.Log(nodename + nodeID.ToString() + " could not be found in the loaded .dyn");
+                            }
+
+                        }
+
+                        loadedStateSet.LoadStateFromXml(name, des, nodes, deserialzedNodes, stateID);
+                    }
+                }
+            }
+            return loadedStateSet;
+        }
+        #endregion
+
+        #region public methods
+        /// <summary>
+        /// method to create and add a new state to this presets collection
+        /// </summary>
+        public void CreateNewState(string name, string description, IEnumerable<NodeModel> currentSelection, Guid id = new Guid())
+        {
+            var inputs = currentSelection;
+            var newstate = new PresetState(name, description, inputs, id);
+            designStates.Add(newstate);
+        }
+
+        public void RemoveState(PresetState state)
```

Comment:
What happens if the `PresetState` is not contained within the collection.  You should check for this.